compiler/integration-tests/test/integration/browser/compile_prove_verify.test.ts

```diff
@@ -159,7 +182,23 @@ test_cases.forEach((testInfo) => {
           isRecursive,
         );
 
-        expect(verified).to.be.true;
+        expect(verified, "Proof fails verification in JS").to.be.true;
+
+        try {
+          let result;
+          if (testInfo.publicInputsLength === 0) {
+            result = await contract.verify(proof, []);
+          } else {
+            const publicInputs = proof.slice(0, testInfo.publicInputsLength);
+            const slicedProof = proof.slice(testInfo.publicInputsLength);
+            result = await contract.verify(slicedProof, [publicInputs]);
```

Comment:
This doesn't look correct. What you're passing is an array with a single element which is a `Uint8Array` (which in this case is 32 bytes long). The verifier contract expects a `bytes32[]` so this works only if we have a single public input. If we used a circuit with two public inputs then this will start failing as you'll try writing 64 bytes into a `bytes32`.

We need a `Uint8Array[]` rather than a `[Uint8Array]`